### PR TITLE
:bug: :apple: :penguin: Fix pasting files containing asterisks

### LINKED ISSUE #547: Uncaught SyntaxError: Invalid regular expression: /^/home/fabian/Projects/testing/python/foo/foo*...
1. Create a folder (Here: `foo`)
2. Create a file (Here: `foo/foo**.txt`)
3. Cut file
4. Paste file to project root

**Atom Version**: 1.0.4-fbe55c3
**System**: linux 4.1.2-2-ARCH
**Thrown From**: [tree-view](https://github.com/atom/tree-view) package, v0.181.1

### Stack Trace

Uncaught SyntaxError: Invalid regular expression: /^/home/fabian/Projects/testing/python/foo/foo**.txt/: Nothing to repeat

```
At /usr/local/share/atom/resources/app.asar/node_modules/tree-view/lib/tree-view.js:990

SyntaxError: Invalid regular expression: /^/home/fabian/Projects/testing/python/foo/foo**.txt/: Nothing to repeat
  at new RegExp (native)
  at TreeView.module.exports.TreeView.pasteEntries (/usr/local/share/atom/resources/app.asar/node_modules/tree-view/lib/tree-view.js:990:61)
  at space-pen-div.atom.commands.add.tree-view:paste (/usr/local/share/atom/resources/app.asar/node_modules/tree-view/lib/tree-view.js:277:26)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/local/share/atom/resources/app.asar/src/command-registry.js:241:29)
  at CommandRegistry.handleCommandEvent (/usr/local/share/atom/resources/app.asar/src/command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (/usr/local/share/atom/resources/app.asar/src/command-registry.js:156:19)
  at [object Object].jQuery.fn.trigger (/usr/local/share/atom/resources/app.asar/src/space-pen-extensions.js:64:23)
  at EventEmitter.<anonymous> (/usr/local/share/atom/resources/app.asar/src/window-event-handler.js:75:67)
  at emitTwo (events.js:87:13)
  at EventEmitter.emit (events.js:169:7)
  at EventEmitter.ipc.sendSync (/usr/local/share/atom/resources/atom.asar/renderer/api/lib/ipc.js:21:31)
  at BrowserWindow.RemoteMemberFunction [as emit] (/usr/local/share/atom/resources/atom.asar/renderer/api/lib/remote.js:110:29)
  at ContextMenuManager.module.exports.ContextMenuManager.showForEvent (/usr/local/share/atom/resources/app.asar/src/context-menu-manager.js:170:31)
  at HTMLDocument.<anonymous> (/usr/local/share/atom/resources/app.asar/src/window-event-handler.js:150:33)
  at HTMLDocument.handler (/usr/local/share/atom/resources/app.asar/src/space-pen-extensions.js:112:34)
  at HTMLDocument.jQuery.event.dispatch (/usr/local/share/atom/resources/app.asar/node_modules/space-pen/vendor/jquery.js:4681:9)
  at HTMLDocument.elemData.handle (/usr/local/share/atom/resources/app.asar/node_modules/space-pen/vendor/jquery.js:4359:46)

```

### Commands

```
     -5:55.6.0 core:confirm (atom-text-editor.editor.mini.is-focused)
  2x -5:42.1.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -5:38.8.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -5:32.3.0 core:move-left (atom-text-editor.editor.mini.is-focused)
  4x -5:32.1.0 core:move-right (atom-text-editor.editor.mini.is-focused)
  4x -5:31.3.0 core:backspace (atom-text-editor.editor.mini.is-focused)
  3x -5:30.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -5:05.8.0 core:move-right (atom-text-editor.editor.mini.is-focused)
     -5:05.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -5:01.5.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -4:47.1.0 core:move-right (atom-text-editor.editor.mini.is-focused)
     -4:46.2.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -4:22.6.0 core:move-right (atom-text-editor.editor.mini.is-focused)
     -4:21.9.0 core:backspace (atom-text-editor.editor.mini.is-focused)
     -4:21.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:43.6.0 settings-view:open (a.btn-issue.btn.btn-error)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "merge-conflicts",
      "open-on-github",
      "atom-cli-diff",
      "status-bar",
      "tool-bar",
      "tool-bar-main",
      "editor-stats",
      "regex-railroad-diagram",
      "spell-check",
      "metrics",
      "color-picker",
      "welcome",
      "linter",
      "linter-coffeescript",
      "linter-pylint"
    ],
    "themes": [
      "unity-ui",
      "solarized-light-syntax"
    ],
    "followSymlinks": true,
    "autoHideMenuBar": true,
    "ignoredNames": [
      ".git",
      "*.mar",
      "*.mkv",
      "*.ts"
    ]
  },
  "tree-view": {}
}
```

### Installed Packages

```coffee
# User
autocomplete-python, v0.5.3
background-tips, v0.25.0
file-icons, v1.5.8
git-log, v0.4.1
linter-coffeelint, v1.0.1
make-executable, v0.1.1
minimap, v4.12.1
package-switch, v0.3.0
project-manager, v1.15.11
unity-ui, v2.0.11

# Dev
No dev packages
```

### PR DESCRIPTION
**Pull request by Tyriar**
A regex was testing to ensure that the pasted file's new path is not part of
the file's original path. Move to use String.prototype.startsWith instead of
trying to sanitize the path as a regex.

Fixes #547

### CODE REVIEW

File: spec/tree-view-spec.coffee [hunk 1]
```diff
@@ -1326,6 +1327,17 @@ describe "TreeView", ->
             atom.commands.dispatch(treeView.element, "tree-view:paste")
             expect(fs.existsSync(path.join(dirPath2, path.basename(filePath4)))).toBeTruthy()
 
+        describe "when pasting a file with an asterisk char '*' in to different directory", ->
+          it "should successfully move the file", ->
+            # Files cannot contain asterisks on Windows
+            return if process.platform.startsWith("win")
```
Comment (50Wliu) [inline spec/tree-view-spec.coffee]: This should be `process.platform is 'win32'`

**Comment by Tyriar** [comment]
@50Wliu is that CI error something to be concerned about? I saw that recently in a package I created that tries to use the same build script https://travis-ci.org/Tyriar/navigation-history

**Comment by 50Wliu** [comment]
Nope, not at all!  Just another network connectivity issue.  I've restarted the build (though it'll take some time to complete - there's a small backlog of Atom builds that Travis is working through right now).

**Comment by kevinsawicki** [comment]
Thanks for fixing this :shipit:

**Pull request merged**